lib/ipaddr_validator.rb

```diff
@@ -23,7 +23,12 @@ def default_options
     private
 
     def validate_single_ipaddr(string, options)
-      ip = IPAddr.new(string)
+      ip =
+        if string.is_a?(IPAddr)
+          string
+        else
+           IPAddr.new(string)
```

Comment:
Please rename the `string` parameter as something appropriate. It can be a `IPAddr` object :-)